### DISCUSSION TITLE
Change language with SEB on MacOS

### DISCUSSION THREAD

**Discussion by ilariacarrara** [SafeExamBrowser/seb-mac#84] [Q&A] [answered]
I study languages at university. I have had an exam of arabic language using Safe Exam Browser. Our professor said that there wouldn't have been any problems in changing the language if we had added arabic to the choices in our computer, which I did. Other students - who had Windows - saw the options and had the possibility to change the language during the exam, while I didn't with my MacOS. Does anyone have some advice?

**Comment by danschlet** [comment, marked as answer]
SEB 2.3, when running on macOS 10.15.4 or later uses the new built-in macOS single app mode (called "Assessment Mode" or AAC). In the Assessment Mode, the macOS menu bar is always invisible. Therefore you cannot use the menu bar "Input menu" to switch the keyboards while SEB is running. The keyboard shortcut (the default shortcut is control - space) for switching the keyboard layout is still working though, you just don't get any visual feedback that the keyboard was switched.

In a later SEB for macOS update we will add a keyboard language control to the SEB Dock, similar like in the Windows version, but we unfortunately don't have enough developer resources and have to prioritize other features.

Your university could also change/add one setting to the SEB configuration file used for your exam to use the old SEB single app kiosk mode also on newer macOS versions, see below. There the menu bar with the keyboard input menu is visible.

Optionally, the old SEB kiosk mode can be used also on latest macOS versions (see Preferences / Security / "Use Assessment Mode (AAC) if available", settings key enableAAC), as AAC doesn't support optionally enabling features like like screen capture/sharing, Siri and Dictation. Assessment Mode also cannot be used if you want to run a video conferencing solution in the background during an exam for proctoring.

**Reply by ilariacarrara** [reply]
Thank you very much for your help!!
